### ISSUE TITLE
Some questions...

### ISSUE THREAD

**Issue by biapar** [chkr1011/MQTTnet#177]
Hi,
I've some questions. 
The scenario is: a server, multiple clients ( device ) with a their id, a control panel for the customer. A customer "see" only his devices.

1 - how to communicate and send message to a client from a "control panel"?
2 - how to receive the messages from my only clients?
3 - Is't possible to send "binary" data like images and so?

Sorry and thanks for you help.

**Comment by VladimirAkopyan** [comment]
It seems like you need to read up on MQTT api design. 
I will do my best to explain from my understanding of what you are trying to accomplish. 
One way to do this is to assign each user assigned a topic: 
`<username>/<device_id>/<message_type>`
The broker implements permissions - check out the wiki for intercepting subscriptions. So you can use them as follows:
* A user is only allowed to subscribe to his own topic. Bob will subscribe to
`Bob/#` and receive messages from `Bob/device_2` and `Bob/device_99`
* A device is only allowed to publish it the topic with the right Username and Device_Id
You might need a more sophisticated topics if you need more functionality

You can send binary data, but MQTT isn't meant to images, it's meant for many small messages. I am not sure what's the max size of payload is in this broker, and if you really-really want to, you could break the image down into pieces and send it over. 
Some guidance: https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices

**Comment by biapar** [comment]
Thanks.
I'm reading about MQTT.
I don't know user in advance. Maybe the a new customer register it to the service and then put in his devices id. On device he configure the IP broker, password and login.
How to send command from the dashboard? Will be the dashboard a new client of Bob topics?

**Comment by chkr1011** [comment]
Yes you can have both. But there is no need if you have all the code running in one process because the server is also able to publish messages. So you don't need a loopback client.
